bigtable-dataflow-parent/bigtable-beam-import/src/test/java/com/google/cloud/bigtable/beam/it/CloudBigtableBeamITTest.java

```diff
@@ -0,0 +1,283 @@
+      new DoFn<String, Mutation>() {
+
+        private static final long serialVersionUID = 1L;
+
+        private Counter rowCounter =
+            Metrics.counter(CloudBigtableBeamITTest.class, "sent_puts");
+
+        @ProcessElement
+        public void processElement(ProcessContext context) throws Exception {
+          String prefix = context.element() + "_";
+          int max = (int) (TOTAL_ROW_COUNT / PREFIX_COUNT);
+          for (int i = 0; i < max; i++) {
+            rowCounter.inc();
+            context.output(new Put(Bytes.toBytes(prefix + i))
+                .addColumn(FAMILY, QUALIFIER, createRandomValue()));
+          }
+        }
+      };
+
+  private Pipeline writeToBigtable(){
+    DataflowPipelineOptions options = createOptions();
+    options.setAppName("testWriteToBigtable-" + System.currentTimeMillis());
+    LOG.info("Started writeToBigtable test with jobName as: %s", options.getAppName());
+
+    CloudBigtableTableConfiguration config =
+        new CloudBigtableTableConfiguration.Builder()
+            .withProjectId(projectId)
+            .withInstanceId(instanceId)
+            .withTableId(TABLE_NAME.getNameAsString())
+            .withConfiguration(BIGTABLE_ADMIN_HOST_KEY, adminEndpoint)
+            .withConfiguration(BIGTABLE_HOST_KEY, dataEndpoint)
+            .build();
+
+    Pipeline p = Pipeline.create(options);
+
+    List<String> keys = new ArrayList<>();
+    for (int i = 0; i < PREFIX_COUNT; i++) {
+      keys.add(RandomStringUtils.randomAlphanumeric(10));
+    }
+    p.apply("Keys", Create.of(keys))
+        .apply("Create Puts", ParDo.of(WRITE_ONE_TENTH_PERCENT))
+        .apply("Write to BT", CloudBigtableIO.writeToTable(config));
+    return p;
+  }
+
+  private Pipeline readFromBigtable(){
+    PipelineOptions options = createOptions();
+    options.setJobName("testReadFromBigtable-" + System.currentTimeMillis());
+    LOG.info("Started readFromBigtable test with jobName as: %s", options.getJobName());
+
+    Scan scan = new Scan();
+    scan.setFilter(new FirstKeyOnlyFilter());
+
+    CloudBigtableScanConfiguration config =
+        new CloudBigtableScanConfiguration.Builder()
+            .withProjectId(projectId)
+            .withInstanceId(instanceId)
+            .withTableId(TABLE_NAME.getNameAsString())
+            .withScan(scan)
+            .withConfiguration(BIGTABLE_ADMIN_HOST_KEY, adminEndpoint)
+            .withConfiguration(BIGTABLE_HOST_KEY, dataEndpoint)
+            .build();
+
+    Pipeline p = Pipeline.create(options);
+    PCollection<Long> count = p
+        .apply("Read from BT", Read.from(CloudBigtableIO.read(config)))
+        .apply("Count", Count.<Result>globally());
+
+    PAssert.thatSingleton(count).isEqualTo(TOTAL_ROW_COUNT);
+    return p;
+  }
+
+  @Test
+  public void testBeamReadAndWrite() throws Exception {
+    // Created Executors for read jobs to run in parallel.
+    ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
Reason for adding executor is to be able to run `BigtableIO.Read` pipeline in parallel.
Currently facing Shaded Library related issue. Would add it to here once resolved.

Reply:
Please don't use an executor.  Please run the write test, wait for it to complete, and then run the read test, or tests.  There's no need for an executor with a single write, and then a single read.  Once we have the multiple reads, you still don't need an executor, of you start both jobs by calling run on each pipeline, and then waiting for the two tests to finish.

Reply:
Then we have to make sure read test always run after write test completes. It would not be independent that's the reason I am running write & read in a single test case.

Reply:
The tests should be dependent on each other.

Reply:
The way you wrote it, you're basically waiting until the write is complete before you do the read test.